### ISSUE TITLE
Write tests for invalid JPEG image files

### ISSUE THREAD

**Issue by leni1** [mozilla-services/screenshots#4395]
Write tests to cover the `assertPng` `invalid-data-url`, `invalid-data-image`, and `invalid-data-image-decoded` paths in cases where the image submitted is `JPEG` rather than `PNG` format.

https://github.com/mozilla-services/screenshots/blob/master/server/src/servershot.js#L21-L53

The tests should hit the `PUT /data/:id/:domain route`.
References: #3482 and the discussion in pull request #4298

**Comment by g-k** [comment]
>  In addition to that, what would be a valid JPEG image since the constant used in the assertPng function is "/9g=" yet the sample JPEG image we have in example_images uses /9j/ as the header. If I were to test this, would I be assuming that a valid JPEG image begins with /9 or it is more fine-grained than that?

https://github.com/mozilla-services/screenshots/pull/4298#issuecomment-387290482

Great question!

We're validating that the image begins with the [SOI bytes 0xFF, 0xD8](https://en.wikipedia.org/wiki/JPEG#Syntax_and_structure) on [line 48](https://github.com/mozilla-services/screenshots/blob/master/server/src/servershot.js#L48) and comparing the buffers of based 64 decoded strings to check this.

> would I be assuming that a valid JPEG image begins with /9 or it is more fine-grained than that?

This is close. We'll want to check the second byte too.

```node
node
> Buffer.from("/9g=", "base64")  // valid JPEG header
<Buffer ff d8>
> Buffer.from("/9j/", "base64") // valid sample JPEG header
<Buffer ff d8 ff>
> Buffer.from("/9", "base64")
<Buffer ff>
```

**Comment by leni1** [comment]
Hello. Would like to work on this issue. Will start once pull request #4298 is merged into master.

**Comment by g-k** [comment]
All yours @leni1

**Comment by ianb** [comment]
This issue is being closed (without being fixed) because [we are discontinuing the Screenshots server](https://blog.mozilla.org/futurereleases/2019/01/24/clarifying-the-future-of-firefox-screenshots/).
